pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,151 @@ package zonemgr
-	userNameserverName string
-	userNameserverIP   string
-	userSubdomain      string
-
-	domainName string
+	soaSerial      int
+	adminEmail     string
+	nameServerName string
+	nameServerIP   string
+	domain         string
+
 	headerPref string
 	headerSuf  string
 
 	header   string
 	aRecords string
 	content  string
 
-	vmiRecordsMap map[string]map[string]ARecord // key=vmi_name+namespace, value=map[key=iface_name, value=ip]
-	//ifacesMap map[string]string
+	vmiRecordsMap map[string][]string // key=vmi_name+namespace, value=ARecords array
 }
 
-func NewZoneFile(clusterName string, userAdminEmail string, userNameserverName string,
-	userNameserverIP string, userSubdomain string) *ZoneFile {
+func NewZoneFile(nameServerIP string, domain string) *ZoneFile {
 	return &ZoneFile{
-		clusterName:        clusterName,
-		userAdminEmail:     userAdminEmail,
-		userNameserverName: userNameserverName,
-		userNameserverIP:   userNameserverIP,
-		userSubdomain:      userSubdomain,
+		nameServerIP: nameServerIP,
+		domain:       domain,
 	}
 }
 
 func (zoneFile *ZoneFile) init() {
-	zoneFile.setDefaultValues()
+	zoneFile.initCustomFields()
 	//before SOA serial
 	zoneFile.generateHeaderPrefix()
 	//after SOA serial
 	zoneFile.generateHeaderSuffix()
 	zoneFile.soaSerial = 0 // TODO make default RANDOM: time.Now().UnixNano()
 	zoneFile.header = zoneFile.generateHeader()
 	zoneFile.content = zoneFile.header
-	zoneFile.vmiRecordsMap = make(map[string]map[string]ARecord)
+	zoneFile.vmiRecordsMap = make(map[string][]string)
 }
 
-func (zoneFile *ZoneFile) setDefaultValues() {
-	zoneFile.domainName = fmt.Sprintf("%s.secondary", zoneFile.clusterName)
-
-	if zoneFile.userSubdomain != "" {
-		zoneFile.domainName = fmt.Sprintf("%s.%s", zoneFile.domainName, zoneFile.userSubdomain)
-	}
-
-	if zoneFile.userNameserverName == "" {
-		zoneFile.userNameserverName = fmt.Sprintf("ns1.%s", zoneFile.domainName)
-	}
-	/*
-		if zoneFile.userNameserverIP == "" {
-			//TODO default???
-		}
-	*/
-	if zoneFile.userAdminEmail == "" {
-		zoneFile.userAdminEmail = fmt.Sprintf("admin.%s", zoneFile.domainName)
+func (zoneFile *ZoneFile) initCustomFields() {
+	if zoneFile.domain == "" {
+		zoneFile.domain = domainDefault
+	} else {
+		zoneFile.domain = fmt.Sprintf("%s.%s", domainDefault, zoneFile.domain)
 	}
+	zoneFile.nameServerName = fmt.Sprintf("%s.%s", nameServerDefault, zoneFile.domain)
+	zoneFile.adminEmail = fmt.Sprintf("%s.%s", adminEmailDefault, zoneFile.domain)
 }
 
-/*
-$ORIGIN mycluster.secondary.mydomain.com
-$TTL 1750
-@       IN      SOA ns1.mycluster.secondary.mydomain.com admin@secdns.com (
-*/
 func (zoneFile *ZoneFile) generateHeaderPrefix() {
-	zoneFile.headerPref = fmt.Sprintf("\n$ORIGIN %s \n$TTL %s \n@ IN SOA %s %s (", zoneFile.domainName, ttl,
-		zoneFile.userNameserverName, zoneFile.userAdminEmail)
-
-	//fmt.Println("======= prefix:\n", zoneFile.headerPref)
-
+	zoneFile.headerPref = fmt.Sprintf("$ORIGIN %s. \n$TTL %s \n@ IN SOA %s. %s. (", zoneFile.domain, ttl,
```

Comment:
plese move the `(` as part of the `generateHeaderSuffix` if possible

Reply:
it's problematic, the idea of dividing header string into headerPrefix and headerSuffix is to be able to change SOA serial and easily regenerate the whole header, and SOA serial value is places after the "(".

Reply:
I am not sure i understand why it is problematic
the '(` is the last character of the `headerPrefix` so it can be simply the first character of the `headerSuffix`
where you are recreating each time the `headerSuffix` again (the serial will be there, and it will be after the `(`)
and writing both to the file
the `headerPrefix` will be set only once, whereas the `headerSuffix` will be recreated each time (because the serial is dynamic)
what am i missing ?

Reply:
What I did is slightly different. I don't recreate `headerSuffix` each time. I create `headerPrefix` and `headerSuffix` just once during initialization and save it for later use, while `headerSuffix` does not contain SOA serial just yet.
Then, each time SOA serial is incremented, I regenerate the header value by concatenating: `headerSuffix` + SOASerial + `headerPrefix`.
This way I don't have to regenerate string that never change.

Reply:
I see, well its a matter of taste, imo it is easier to grasp when the `(` are part of of one variable
but i won't insist on it